gridcomps/History3G/HistoryCollectionGridComp_private.F90

```diff
@@ -217,20 +218,37 @@ function get_expression_variables(expression, rc) result(variables)
       _RETURN(_SUCCESS)
    end function get_expression_variables
 
-   function get_current_time_index(initial_time, current_time, frequency) result(time_index)
-      integer :: time_index
+   subroutine get_current_time_info(initial_time, current_time, frequency, time_index, time_vector, rc)
       type(ESMF_Time), intent(in) :: initial_time
       type(ESMF_Time), intent(in) :: current_time
       type(ESMF_TimeInterval), intent(in) :: frequency
+      integer, intent(out) :: time_index
+      real, allocatable, intent(out) :: time_vector(:)
+      integer, intent(out), optional :: rc
 
+      integer :: status,i
       type(ESMF_Time) :: temp_time
+      type(ESMF_TimeInterval) :: tint
+      real(ESMF_KIND_R8) :: time_in_minutes
+
       time_index = 0
       temp_time = initial_time
       do while( temp_time <= current_time)
          temp_time = temp_time + frequency
          time_index = time_index + 1
       enddo
-   end function get_current_time_index
+
+      allocate(time_vector(time_index),_STAT)
+      temp_time = initial_time
+      time_vector(1) = 0
+      do i=2,time_index
+         temp_time = temp_time + frequency
+         tint = temp_time - initial_time
+         call ESMF_TimeIntervalGet(tint, m_r8=time_in_minutes, _RC)
```

Comment:
The arbitrariness of "minutes" (vs seconds) catches my eye here.  Does GEOS only use "time-since" in units of minutes?    I.e., do we need a generalization of this?